src/test/java/uk/gov/hmcts/ccd/ElasticsearchIT.java

```diff
@@ -0,0 +1,1447 @@
+import static uk.gov.hmcts.ccd.test.ElasticsearchTestHelper.PHONE_VALUE;
+import static uk.gov.hmcts.ccd.test.ElasticsearchTestHelper.PHONE_VALUE_WITH_SPACE;
+import static uk.gov.hmcts.ccd.test.ElasticsearchTestHelper.POST_CODE_FIELD;
+import static uk.gov.hmcts.ccd.test.ElasticsearchTestHelper.POST_CODE_VALUE;
+import static uk.gov.hmcts.ccd.test.ElasticsearchTestHelper.STATE;
+import static uk.gov.hmcts.ccd.test.ElasticsearchTestHelper.STATE_VALUE;
+import static uk.gov.hmcts.ccd.test.ElasticsearchTestHelper.STREET_VALUE;
+import static uk.gov.hmcts.ccd.test.ElasticsearchTestHelper.TEXT_ALIAS;
+import static uk.gov.hmcts.ccd.test.ElasticsearchTestHelper.TEXT_AREA_FIELD;
+import static uk.gov.hmcts.ccd.test.ElasticsearchTestHelper.TEXT_AREA_VALUE;
+import static uk.gov.hmcts.ccd.test.ElasticsearchTestHelper.TEXT_FIELD;
+import static uk.gov.hmcts.ccd.test.ElasticsearchTestHelper.TEXT_VALUE;
+import static uk.gov.hmcts.ccd.test.ElasticsearchTestHelper.TOWN_FIELD;
+import static uk.gov.hmcts.ccd.test.ElasticsearchTestHelper.TOWN_VALUE;
+import static uk.gov.hmcts.ccd.test.ElasticsearchTestHelper.VALUE_SUFFIX;
+import static uk.gov.hmcts.ccd.test.ElasticsearchTestHelper.YES_OR_NO_FIELD;
+import static uk.gov.hmcts.ccd.test.ElasticsearchTestHelper.YES_OR_NO_VALUE;
+import static uk.gov.hmcts.ccd.test.ElasticsearchTestHelper.alias;
+import static uk.gov.hmcts.ccd.test.ElasticsearchTestHelper.caseData;
+import static uk.gov.hmcts.ccd.test.ElasticsearchTestHelper.caseTypesParam;
+import static uk.gov.hmcts.ccd.test.ElasticsearchTestHelper.createPostRequest;
+
+@RunWith(Enclosed.class)
+public class ElasticsearchIT extends ElasticsearchBaseTest {
+
+    private static final String DATA_DIR = "elasticsearch/data";
+
+    @Inject
+    private WebApplicationContext wac;
+
+    @Mock
+    private Authentication authentication;
+
+    @Mock
+    private SecurityContext securityContext;
+
+    @Inject
+    private ApplicationParams applicationParams;
+
+    private MockMvc mockMvc;
+
+    @BeforeAll
+    public static void initElastic(@Autowired EmbeddedElastic embeddedElastic) throws IOException,
+        InterruptedException {
+        embeddedElastic.start();
+        initData(embeddedElastic);
+    }
+
+    private static void initData(EmbeddedElastic embeddedElastic) throws IOException {
+        PathMatchingResourcePatternResolver resourceResolver = new PathMatchingResourcePatternResolver();
+        for (String idx : INDICES) {
+            Resource[] resources =
+                resourceResolver.getResources(String.format("classpath:%s/%s/*.json", DATA_DIR, idx));
+            for (Resource resource : resources) {
+                String caseString = IOUtils.toString(resource.getInputStream(), UTF_8);
+                embeddedElastic.index(idx, INDEX_TYPE, caseString);
+            }
+        }
+    }
+
+    @AfterAll
+    public static void tearDownElastic(@Autowired EmbeddedElastic embeddedElastic) {
+        embeddedElastic.stop();
```

Comment:
is it worth putting a LOG message here so we can track elastic search stopping?

Reply:
I think ES has its own logging when it stops, but will add one at this level as well